chemprop/nn/predictors.py

```diff
@@ -120,8 +120,17 @@ def __init__(
     ):
         super().__init__(n_tasks, input_dim, hidden_dim, n_layers, dropout, activation, criterion)
 
-        self.register_buffer("loc", torch.tensor(loc).view(-1, 1))
-        self.register_buffer("scale", torch.tensor(scale).view(-1, 1))
+        if not isinstance(loc, Tensor):
+            loc = torch.ones(1, self.n_tasks) * loc
+        else:
+            loc = torch.tensor(loc).view(1, -1)
+        self.register_buffer("loc", loc)
+
+        if not isinstance(scale, Tensor):
+            scale = torch.ones(1, self.n_tasks) * scale
+        else:
+            scale = torch.tensor(scale).view(1, -1)
+        self.register_buffer("scale", scale)
```

Comment:
I see that the CLI doesn't use give the scaler to `build_model()` which means that when making a `RegressionFFN` the `loc` and `scale` use default values of 0 and 1. Later in the CLI we set these manually (`model.predictor.register_buffer("loc", torch.tensor(scaler.mean_).view(1, -1))`). But if the user was building their own `RegressionFFN`, they would use this code. It looks like this code is written to handle the use case of passing a pair of floats loc/scale to transform multiple tasks. I think if the user has multiple tasks, they will have a different loc/scale for each target and we can expect it to be an array. (So the original code would be fine?)

This also reminds me that we wanted to spend time thinking about how we load and save scalers in general.

Reply:
This part of code is to handle error I ran into when loading a trained multitask model. The following lines initialize the `loc` and `scale` with default values of 0 and 1 and turn them into a 1 by 1 tensor respectively, but the trained model contains a different shape of tensor (1 by number of tasks), so it causes pytorch to throw error when writing in the weights.

```
self.register_buffer("loc", torch.tensor(loc).view(1, -1))
self.register_buffer("scale", torch.tensor(scale).view(1, -1))
```

I see what you're saying. I will change the type check so that it only handles the case when `loc` and `scale` are floats.